open-api/rest-catalog-open-api.yaml

```diff
@@ -2800,6 +3053,59 @@ components:
           additionalProperties:
             type: string
 
+    PlanContext:
+      type: object
+      required:
+        - select
+        - options
+      properties:
+        select:
+          description: A list of the selected columns
+          type: array
+          items:
+            type: string
+        project:
+          description: A list of the projections
+          type: array
+          items:
+            type: string
+        filter:
+          $ref: '#/components/schemas/Expression'
+        options:
+          type: object
+          additionalProperties:
+            type: string
+        use-snapshot:
+          type: integer
+          format: int64
+        from-snapshot-id:
+          type: integer
+          format: int64
+        to-snapshot-id:
+          type: integer
+          format: int64
+        as-of-time:
+          type: integer
+          format: int64
+        case-sensitive:
+          type: boolean
+        use-ref:
+          type: string
+
+    PreplanTableRequest:
+      type: object
+      properties:
+        allOf:
+          $ref: '#/components/schemas/PlanContext'
+
+    PlanTableRequest:
+      type: object
+      properties:
+        allOf:
+          $ref: '#/components/schemas/PlanContext'
+        plan-task:
```

Comment:
I think that `plan-task` is incompatible with some of the `PlanContext` arguments. If `plan-task` is present, then the version selection arguments (`snapshot-id` and `snapshot-range`) must be omitted. I'm not sure it makes sense to use a shared set of arguments like this because of the changes to behavior that need to be documented with the extra option.

Reply:
Thanks for catching this, will amend this